types/jest/index.d.ts

```diff
@@ -198,7 +197,9 @@ declare namespace jest {
      *   spy.mockRestore();
      * });
      */
-    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType?: 'get' | 'set'): SpyInstance<T[M]>;
+    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType: 'get'): SpyInstance<T[M], []>;
+    function spyOn<T extends {}, M extends keyof T>(object: T, method: M, accessType: 'set'): SpyInstance<void, [T[M]]>;
+    function spyOn<T extends {}, M extends keyof T>(object: T, method: M): T[M] extends (...args: any[]) => any ? SpyInstance<ReturnType<T[M]>, ArgsType<T[M]>> : never;
```

Comment:
`extends (...args: never[]) => unknown`  is the most compatible function signature

Reply:
This `never` return sucks but there's really no better way, huh... 😢 

`<..., T[M] extends (...args: never[]) => unknown>` would DWIM but 1) it's a syntax error; 2) it's not a real parameter, just an additional constraint, and it can't even be named.